lib/new_relic/agent/system_info.rb

```diff
@@ -128,7 +128,7 @@ def self.parse_cpuinfo(cpuinfo)
 
         num_physical_packages = cores.keys.map(&:first).uniq.size
         num_physical_cores = cores.size
-        num_logical_processors = cores.values.reduce(0, :+)
```

Comment:
This cop is unsafe if sum is called on an object that doesn't have a sum method, such as a String. `cores.values` should always be an array of integers based on the default values of the Hash defined on [line 110](https://github.com/newrelic/newrelic-ruby-agent/pull/1840/commits/02b7aabde79f256521674c65c1287e3d585950b0#diff-f6f8dd00b6df0ce9cdcb61bc2ff7ce94f894a821e859d9c1e0e991c57d2608acL110)